### ISSUE TITLE
K.gradients returning [None]

### ISSUE THREAD

**Issue by vianamp** [keras-team/keras#8478]
tensorflow (1.4.0)
Keras (2.1.0)

Hi. I am running this code https://github.com/jacobgil/keras-grad-cam/blob/master/grad-cam.py and after updating TF and Keras the code is crashing at `grads = normalize(K.gradients(loss, conv_output)[0])`. It says the output of `K.gradients(loss, conv_output)` is none. Any ideas? Code was working fine before.

**Comment by bdwyer2** [comment]
By any chance does it work with Keras 2.1.1?

**Comment by drlego9** [comment]
Experiencing the same problem here. 'K.gradients(loss, conv_output)[0]' returns None after upgrading to tensorflow-gpu (1.4.0) and Keras (2.1.1).

**Comment by bdwyer2** [comment]
This may be related to #8481

**Comment by vianamp** [comment]
Tried to update to Keras 2.1.1 by still experiencing same issue.

**Comment by vianamp** [comment]
@fchollet any ideas here? Do you need more information?

**Comment by vianamp** [comment]
@drlego9 did you solve this?

**Comment by tchlux** [comment]
This is likely rooted in [Tensorflow](https://github.com/tensorflow/tensorflow/issues/783). There isn't a solution other than making sure your network is differentiable from start to end. I don't know how to ensure that for your use case.

**Comment by ParmpalSinghGill** [comment]
initialize the loss by 0 as
loss=0
insted of 
loss = K.variable(0.)
resolve the issue
